lib/auth/accountrecovery_test.go

```diff
@@ -517,6 +517,115 @@ func TestApproveAccountRecovery_WithLock(t *testing.T) {
 	require.Len(t, attempts, 0)
 }
 
+func TestCompleteAccountRecovery(t *testing.T) {
+	srv := newTestTLSServer(t)
+	ctx := context.Background()
+	mockEmitter := &events.MockEmitter{}
+	srv.Auth().emitter = mockEmitter
+
+	defaultModules := modules.GetModules()
+	defer modules.SetModules(defaultModules)
+	modules.SetModules(&testWithCloudModules{})
+
+	u, err := createUserWithSecondFactorAndRecoveryCodes(srv)
+	require.NoError(t, err)
+
+	// Test new password with lock that should not affect changing authn.
+	triggerLoginLock(t, srv.Auth(), u.username)
+
+	// Acquire an approved token for recovering password.
+	approvedToken, err := srv.Auth().createRecoveryToken(ctx, u.username, UserTokenTypeRecoveryApproved, types.UserTokenUsage_USER_TOKEN_RECOVER_PASSWORD)
```

Comment:
I realize it's a bit of a pain, but could we go through the complete flow, beginning with the Start RPC? It can be a different test case if you prefer, so you can focus this one on different aspects of CompleteAccountRecovery.

It's important to test the process end-to-end, with calls similar to what we expect the end user to generate. Plus, it serves as a nice documentation of the process.